src/ExchangeSharp/API/Exchanges/KuCoin/ExchangeKuCoinAPI.cs

```diff
@@ -455,41 +455,36 @@ protected override async Task<ExchangeWithdrawalResponse> OnWithdrawAsync(Exchan
             return response;
         }
 
-        #endregion Private APIs
+		#endregion Private APIs
 
-        #region Websockets
+		#region Websockets
 
-        protected override async Task<IWebSocket> OnGetTickersWebSocketAsync(Action<IReadOnlyCollection<KeyValuePair<string, ExchangeTicker>>> callback, params string[] marketSymbols)
-        {
-            var websocketUrlToken = GetWebsocketBulletToken();
-            return await ConnectWebSocketAsync
-            (
-                $"?token={websocketUrlToken}&acceptUserMessage=true", async (_socket, msg) =>
-                {
-                    JToken token = JToken.Parse(msg.ToStringFromUTF8());
-                    if (token["type"].ToStringInvariant() == "message")
-                    {
-						const string topicPrefix = "/market/ticker:";
+		protected override async Task<IWebSocket> OnGetTickersWebSocketAsync(Action<IReadOnlyCollection<KeyValuePair<string, ExchangeTicker>>> callback, params string[] marketSymbols)
+		{
+			var websocketUrlToken = GetWebsocketBulletToken();
+			return await ConnectWebSocketAsync
+			(
+				$"?token={websocketUrlToken}&acceptUserMessage=true", async (_socket, msg) =>
+				{
+					//{"data":{"sequence":"1612818290539","bestAsk":"0.387788","size":"304.072","bestBidSize":"654.3404","price":"0.387788","time":1618870257524,"bestAskSize":"0.00005226","bestBid":"0.386912"},"subject":"XEM-USDT","topic":"/market/ticker:all","type":"message"}
+					JToken token = JToken.Parse(msg.ToStringFromUTF8());
+					if (token["type"].ToStringInvariant() == "message")
+					{
 						var dataToken = token["data"];
-                        var marketSymbol = token["topic"].ToStringInvariant();
-						if (!marketSymbol.StartsWith(topicPrefix))
-							return;
-						marketSymbol = marketSymbol.Substring(topicPrefix.Length);
+						var marketSymbol = token["subject"].ToStringInvariant();
 						ExchangeTicker ticker = await this.ParseTickerAsync(dataToken, marketSymbol,
 							"bestAsk", "bestBid", "price", "size", null, "time", TimestampType.UnixMilliseconds, idKey: "sequence");
 						callback(new List<KeyValuePair<string, ExchangeTicker>>() { new KeyValuePair<string, ExchangeTicker>(marketSymbol, ticker) });
-                    }
-                }, async (_socket) =>
-                {
-                    //need to subscribe to tickers one by one
-                    marketSymbols = marketSymbols == null || marketSymbols.Length == 0 ? (await GetMarketSymbolsAsync()).ToArray() : marketSymbols;
```

Comment:
Your new code subscribing to "all" is good. However, can you keep the functionality where if the user specifies certain `marketSymbols`, then only those symbols are subscribed to?